server/game/costs.js

```diff
@@ -239,6 +239,43 @@ const Costs = {
         };
     },
     /**
+     * Cost that requires discarding a card from hand.
+     */
+    discardFromHand: function() {
+        return {
+            canPay: function(context) {
+                return context.player.hand.size() >= 1;
+            },
+            resolve: function(context) {
+                var result = {
+                    resolved: false
+                };
+
+                context.game.promptForSelect(context.player, {
+                    cardCondition: card => card.location === 'hand',
+                    activePromptTitle: 'Select card to discard',
+                    source: context.source,
+                    onSelect: (player, card) => {
+                        context.discardCostCard = card;
+                        result.value = true;
+                        result.resolved = true;
+
+                        return true;
+                    },
+                    onCancel: () => {
+                        result.value = false;
+                        result.resolved = true;
+                    }
+                });
+
+                return result;
+            },
+            pay: function(context) {
+                context.player.moveCard(context.discardCostCard, 'discard pile');
```

Comment:
This should be done explicitly with `player.discardCard` to ensure the appropriate discard events are raised (e.g. LoCR Cersei should be able to react to you paying this cost).